frontend/src/pages/expenses/Expenses.tsx

```diff
@@ -1,19 +1,129 @@
 import React, { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
 import Button from "../../components/button/button";
 import Hero from "../../components/hero/hero";
+import Loader from "../../components/loader/loader";
+import Table, { ITableHead } from "../../components/table/table";
+import { TAddiotinalLabel } from "../../components/table/table.header";
+import monthNames from "../../constants/months";
+import formatCurrency from "../../utils/formatCurrency";
+import formatDate from "../../utils/formatDate";
+
+interface IExpenseOutput extends Omit<IExpense, "date" | "amount" | "_id"> {
+  _id: string;
+  actions: JSX.Element;
+  date: string;
+  amount: string;
+}
+
+interface IExpensesPerMonth {
+  month: number;
+  total: number;
+  year: number;
+  rows: any[];
+}
 
 const Expenses = (): JSX.Element => {
-  return (
+  const [expensesRaw, setExpensesRaw] = useState<IExpense[] | null>(null);
+  const [expenses, setExpenses] = useState<IExpensesPerMonth[]>([]);
+
+  useEffect(() => {
+    const fetchExpenses = async () => {
+      const rawExpenses = await fetch("/api/expense");
+      setExpensesRaw(await rawExpenses.json());
+    };
+    fetchExpenses();
+  }, []);
+
+  useEffect(() => {
+    if (expensesRaw === null) return;
+
+    setExpenses(
+      expensesRaw.reduce<IExpensesPerMonth[]>(
```

Comment:
This one should be moved to generic code at some point... both expense and income pages are going to use same logic...